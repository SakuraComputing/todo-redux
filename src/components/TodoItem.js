import React from 'react';

const TodoItem = ({ todo }) => {
    return (
        <div className="todo-list-item">
            <h3>{todo}</h3>
        </div>
    );
}

export default TodoItem;