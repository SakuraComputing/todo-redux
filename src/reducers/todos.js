const todoReducerDefaultState = [];

export default (state = todoReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_TODO':
            return [
                ...state,
                action.todo
            ];
        case 'SET_TODOS':
            return action.todos;
        default:
            return state;
    }
};