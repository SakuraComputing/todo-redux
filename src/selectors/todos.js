export default (todos, { text, sortBy }) => {
    console.log('Is it getting in here?')
    return todos.filter((todo) => {
        const textMatch = todo.todo.toLowerCase().includes(text.toLowerCase());
        return textMatch;
    }).sort((a, b) => {
        if (sortBy === 'Asc') {
            return a.todo - b.todo;
        } else {
            return b.todo - a.todo;
        }
    });
};
