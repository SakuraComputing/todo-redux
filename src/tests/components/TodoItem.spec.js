import React from 'react';
import { shallow } from 'enzyme';
import todos from '../fixtures/todos';
import TodoItem from '../../components/TodoItem';

describe('Todo List Item Component', () => {
    it('should render Todo list item',() => {
       const wrapper = shallow(<TodoItem {...todos[0]}  />) ;
       expect(wrapper).toMatchSnapshot();
    });    
})

