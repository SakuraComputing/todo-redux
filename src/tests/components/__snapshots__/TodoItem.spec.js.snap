// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo List Item Component should render Todo list item 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoItem
    id={0}
    todo="Setup Stands"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h3>
        Setup Stands
      </h3>,
      "className": "todo-list-item",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Setup Stands",
      },
      "ref": null,
      "rendered": "Setup Stands",
      "type": "h3",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3>
          Setup Stands
        </h3>,
        "className": "todo-list-item",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Setup Stands",
        },
        "ref": null,
        "rendered": "Setup Stands",
        "type": "h3",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
